extensions/qcom/GLSL_QCOM_image_processing2.txt

```diff
@@ -0,0 +1,212 @@
+Name
+
+    QCOM_image_processing2
+
+Name Strings
+
+    GL_QCOM_image_processing2
+
+Contact
+
+    Wooyoung Kim, Qualcomm Technologies Inc.  (wooykim 'at' qti.qualcomm.com)
+
+Contributors
+
+    Wooyoung Kim, Qualcomm Technologies Inc.
+    Jeff Leger,   Qualcomm Technologies Inc.
+    Ruihao Zhang, Qualcomm Technologies Inc.
+
+Status
+
+    Final
+
+Version
+
+    Last Modified Date: 22-Jan-2024
+    Revision: 1
+
+Number
+
+    TBD.
+
+Dependencies
+
+    This extension can be applied to OpenGL GLSL versions 1.40
+    (#version 140) and higher.
+
+    This extension can be applied to OpenGL ES ESSL versions 3.10
+    (#version 310) and higher.
+
+    Requires GL_KHR_vulkan_glsl.
+
+    Requires GL_QCOM_image_processing.
+
+
+Overview
+
+    This extension is a follow-on to an earlier extension GL_QCOM_image_processing which has
+    added four new shader instructions to perform some of the frequently sought image
+    processing tasks. This extension adds another four new image processing operations.
+
+    Block Matching Window SSD: The operation repeatedly performs block match operations
+      across a 2D window specified on the sampler used to create the target image to
+      compute an error at each (X,Y) within the 2D window that describes whether the texel
+      in the target image matches the corresponding texel in the reference image and
+      return either the minimum or maximum error using sum of squares (SSD) of the differences. 
+  
+    Block Matching Window SAD: This operation follows the same pattern as "Block Matching
+      Window SSD", computing the sum of the absolute differences (SAD)
+
+    Block Matching Gather SSD: The operation computes errors at 4 texels at the given offsets
+      that describe whether the texels in the target image match the corresponding texels in 
+      the reference image. It returns the SSD error metrics at the 4 texel offsets in the X, Y, 
+      Z, and W components. 
+
+    Block Matching Gather SAD: This operation follows the same pattern as "Block Matching
+     Gather SSD", computing the sum of the absolute differences at 4 offsets.
+
+    Each of the above 4 image processing operations are limited to single-component formats.
+
+
+Modifications to the OpenGL GL Shading Language Specification, Version 1.40, dated 22 November 2009
+
+    Including the following line in a shader can be used to control the
+    language features described in this extension:
+
+        #extension GL_QCOM_image_processing2  : <behavior>
+
+    where <behavior> is as specified in Section 3.3.
+
+    A new preprocessor #define is added to the OpenGL Shading Language:
+
+        #define GL_QCOM_image_processing2    1
+
+    Add to the table at the end of Subsection 8.9.2 "Texel Lookup Functions"
+    the following image processing functions. Note that these built-ins rely on separate
+    samplers and textures as defined in GL_KHR_vulkan_glsl.
```

Comment:
They actually don't seem to rely on separate textures and samplers. I think either this note should be removed or the functions changed to use (texture2D, sampler) pairs instead of sampler2D.